lib/secrets/secrets_test.go

```diff
@@ -95,3 +98,53 @@ func TestDiskSecretStore(t *testing.T) {
 		assert.False(t, exists)
 	})
 }
+
+func TestSubPropertyBased(t *testing.T) {
+	parameters := gopter.DefaultTestParameters()
+	parameters.MinSuccessfulTests = 500
+	properties := gopter.NewProperties(parameters)
+
+	tmpDir, _ := ioutil.TempDir("", "secret-store-tests")
+	defer os.RemoveAll(tmpDir)
+
+	uid := os.Getuid()
+	gid := os.Getgid()
+	dataDir := filepath.Join(tmpDir, "data-dir")
+
+	dataStore := secrets.NewDiskStore(dataDir, uid, gid)
+	err := dataStore.Initialize()
+	require.NoError(t, err, "failed to initialize")
+
+	properties.Property("generate multiple random test cases to check if get and set can roundtrip a secret", prop.ForAll(
+		func(a string, b string, c string) bool {
+			testSecretName := secrets.SecretName{a, b}
+			testSecretContent := []byte(c)
+			err := dataStore.SetSecret(testSecretName, testSecretContent)
+			if err != nil {
+				return reportErrorAndYieldFalse(t, err)
+			}
+			secretContent, err := dataStore.GetSecret(testSecretName)
+			if err != nil {
+				return reportErrorAndYieldFalse(t, err)
+			}
+			return string(testSecretContent) == string(secretContent)
+		},
+		stringGen(10).WithLabel("a"),
+		stringGen(10).WithLabel("b"),
+		stringGen(10).WithLabel("c"),
```

Comment:
I think we want to generate a variety of secrets. We don't want to limit it just 10 alphanumerical characters

Reply:
content will be randomly generated from sizes 1 byte to 5 kb